utils/warnings.ml

```diff
@@ -94,6 +94,75 @@ type t =
   | Unused_functor_parameter of string      (* 67 *)
 ;;
 
+let all =
+  [ Comment_start;
+    Comment_not_end;
+    Fragile_match "";
+    Partial_application;
+    Labels_omitted [];
+    Method_override [];
+    Partial_match "";
+    Non_closed_record_pattern "";
+    Statement_type;
+    Unused_match;
+    Unused_pat;
+    Instance_variable_override [];
+    Illegal_backslash;
+    Implicit_public_methods [];
+    Unerasable_optional_argument;
+    Undeclared_virtual_method "";
+    Not_principal "";
+    Without_principality "";
+    Unused_argument;
+    Nonreturning_statement;
+    Preprocessor "";
+    Useless_record_with;
+    Bad_module_name "";
+    All_clauses_guarded;
+    Unused_var "";
+    Unused_var_strict "";
+    Wildcard_arg_to_constant_constr;
+    Eol_in_string;
+    Duplicate_definitions ("", "", "", "");
+    Multiple_definition ("", "", "");
+    Unused_value_declaration "";
+    Unused_open "";
+    Unused_type_declaration "";
+    Unused_for_index "";
+    Unused_ancestor "";
+    Unused_constructor ("", false, false);
+    Unused_extension ("", false, false, false);
+    Unused_rec_flag;
+    Name_out_of_scope ("", [], false);
+    Ambiguous_name ([], [], false, "");
+    Disambiguated_name "";
+    Nonoptional_label "";
+    Open_shadow_identifier ("", "");
+    Open_shadow_label_constructor ("", "");
+    Bad_env_variable ("", "");
+    Attribute_payload ("", "");
+    Eliminated_optional_arguments [];
+    No_cmi_file ("", None);
+    Bad_docstring false;
+    Expect_tailcall;
+    Fragile_literal_pattern;
+    Misplaced_attribute "";
+    Duplicated_attribute "";
+    Inlining_impossible "";
+    Unreachable_case;
+    Ambiguous_pattern [];
+    No_cmx_file "";
+    Assignment_to_non_mutable_value;
+    Unused_module "";
+    Unboxable_type_in_prim_decl "";
+    Constraint_on_gadt;
+    Erroneous_printed_signature "";
+    Unsafe_without_parsing;
+    Redefining_unit "";
+    Unused_open_bang "";
+    Unused_functor_parameter "" ]
```

Comment:
I think that it would be better to use the Objectively Right indentation style here, which is to use a terminator for the last item and the enclosing bracket on the next line, because we are going to keep adding new stuff to this list.

(What happens when we forget ? Could there maybe be a check somewhere that the last warning number is also the last known warning?)

Reply:
Fixed indentation. To make sure that we are not forgetting anything, one possibility could generate the list `all` and the function `name` with a script.